internal/compiler/langtype.rs

```diff
@@ -797,6 +782,30 @@ impl<'a> PropertyLookupResult<'a> {
     }
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub struct Callback {
+    pub return_type: Option<Type>,
+    pub args: Vec<Type>,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct Function {
+    pub return_type: Type,
+    pub args: Vec<Type>,
+}
```

Comment:
Not sure if there is a point to have two different type for this. Having the same struct may simplify the code a bit maybe.
(Can use Type::Void instead of None for the return_type)

Reply:
True, but I won't spend time on this refactor, I'm really only interested in the performance aspect currently, and refactoring the code more than needed for that purpose is not something I'm interested in right now I'm afraid :)